galaxyproject/galaxy
# Set minimum requirements for starlette-context and fastapi

## What did you do? 
Set minimum Python version for starlette-context to 3.7 (starlette-context versions <=0.1.3 were not enforcing this, but it was never supported on Python <3.7).
Set minimum version of FastAPI to 0.63.0 because all fastapi versions >=0.1.18 pin starlette, and poetry tends to prefer selecting the latest starlette (still excluded by the fastapi pins) and downgrading fastapi to 0.1.17.

## Why did you make this change?
This prevents automatic downgrading of the dependency by Poetry's dependency resolver as a result of executing `make update-dependencies` which runs Poetry's `update` on all dependencies. (On 2021-03-15 Poetry suggested a downgrades for both starlette-context and fastapi)

## How to test the changes? 
- Run `make update-dependencies`
- Check that starlette-context is >= 0.3.1 in requirements.txt
